src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -22,12 +22,30 @@
 
 /**
  * Compiler pass to register tagged services for an event dispatcher.
+ *
+ * @psalm-type BeforeAfterDefinition = string|array{0: string, 1: string}
+ * @psalm-type ListenerDefinition = array{serviceId: string, event: string, method: string, before?: BeforeAfterDefinition, after?: BeforeAfterDefinition, priority?: int, dispatchers: list<string>}
```

Comment:
We don't use psalm type aliases in symfony

Reply:
would you recommend that I use array shapes in PHPDoc everywhere instead?
I was also wondering if I should create an actual class for this, WDYT?

Reply:
I would use the array shape directly.

Anyway, if you follow my other comments, it will remove many usages.

Reply:
done, but now we have a lot of cryptic phpdoc really less understandable by humans :sweat_smile: